openpifpaf/network/losses/composite.py

```diff
@@ -92,6 +103,10 @@ def _localization_loss(self, x_regs, t_regs):
         batch_size = t_regs.shape[0]
 
         reg_losses = []
+        if self.weights is not None:
+            weight = torch.ones_like(t_regs[:, :, 0], requires_grad=False)
+            weight[:] = self.weights
+            weights_stacked = weight.unsqueeze(0).repeat(self.n_vectors, 1, 1, 1, 1)
```

Comment:
Hi @DuncanZauss , I saw you are cleaning things up. Something must have worked :)
While you are cleaning up, can you remove this `repeat`? From the idea of "Composite" fields, there should only ever be one consistent weight across all the components.

In the language of the paper:
<img width="566" alt="image" src="https://user-images.githubusercontent.com/2775078/110100534-cea49a80-7da2-11eb-863e-e6324cb6388c.png">
We want to weigh the training of this object p with one weight per J.

Reply:
Hi Sven,
I created this repeat in case the weighing is also used for fields, where there are multiple vector/scales, e.g. if a CAF should be weight. I was unsure if the autograd would still work if I use the same weight for all vector/scale fields.